packages/hub/tasks/scheduled-payment-on-chain-creation-waiter.ts

```diff
@@ -0,0 +1,95 @@
+import { inject } from '@cardstack/di';
+import * as Sentry from '@sentry/node';
+import { isBefore, subDays } from 'date-fns';
+
+export default class ScheduledPaymentOnChainCreationWaiter {
+  prismaManager = inject('prisma-manager', { as: 'prismaManager' });
+  web3 = inject('web3-http', { as: 'web3' });
+  cardpay = inject('cardpay');
+
+  async perform(payload: { scheduledPaymentId: string }) {
+    let prisma = await this.prismaManager.getClient();
+
+    let scheduledPaymentModule = await this.cardpay.getSDK('ScheduledPaymentModule', this.web3.getInstance());
+    let scheduledPayment = await prisma.scheduledPayment.findFirstOrThrow({
+      where: { id: payload.scheduledPaymentId },
+    });
+
+    if (scheduledPayment.creationBlockNumber) {
+      // Did we somehow spawn this task after it already updated creationBlockNumber previously?
+      return Sentry.captureException(
+        new Error('Task run for scheduled payment that already has a creation block number'),
+        {
+          tags: {
+            action: 'scheduled-payment-task',
```

Comment:
You could set up an “Alert” on Sentry to get email notifications if/when these errors are encountered, then you don’t have to be checking Sentry manually

<img width="328" alt="alert-web-team - cardstack - hub - Sentry 2022-09-28 12-28-48" src="https://user-images.githubusercontent.com/43280/192849554-2405d825-a076-4af8-b5bf-9c727c7c0734.png">